File: frontend/src/data/capabilities.ts
export type Capability = {
  id: string;
  type: string;
  category: string;
  name: string;
  description: string;
};

export const capabilities: Capability[] = [
  {
    id: "data-quality-rules",
    type: "Skill",
    category: "Data Governance",
    name: "Data Quality Rules",
    description:
      "Generate, validate, and execute reusable data quality rules.",
  },
  {
    id: "data-quality-scorecard",
    type: "App",
    category: "Observability",
    name: "Data Quality Scorecard",
    description:
      "Measure completeness, validity, consistency, and uniqueness.",
  },
  {
    id: "data-lineage-explorer",
    type: "Agent",
    category: "Discovery",
    name: "Data Lineage Explorer",
    description:
      "Understand how enterprise data moves between systems.",
  },
  {
    id: "policy-exception-advisor",
    type: "Agent",
    category: "Compliance",
    name: "Policy Exception Advisor",
    description:
      "Review policy exceptions and recommend compliant remediation paths.",
  },
];

File: backend/app/catalogue/capabilities.py
CAPABILITIES = [
    {
        "id": "data-quality-rules",
        "name": "Data Quality Rules",
        "description": (
            "Generate, validate, and execute reusable "
            "data quality rules."
        ),
        "type": "Skill",
        "category": "Data Governance",
    },
    {
        "id": "data-quality-scorecard",
        "name": "Data Quality Scorecard",
        "description": (
            "Measure completeness, validity, consistency, "
            "and uniqueness."
        ),
        "type": "App",
        "category": "Data Observability",
    },
    {
        "id": "data-lineage-explorer",
        "name": "Data Lineage Explorer",
        "description": (
            "Understand how enterprise data moves "
            "between systems."
        ),
        "type": "Agent",
        "category": "Data Intelligence",
    },
]


def get_capability_by_id(capability_id: str):
    return next(
        (
            item
            for item in CAPABILITIES
            if item["id"] == capability_id
        ),
        None,
    )